### DISCUSSION TITLE
Easy way to do jpm build and then run the executable?

### DISCUSSION THREAD

**Discussion by m-r-hunt** [janet-lang/janet#1100] [Q&A] [answered]
I have a jpm project set up to build an executable. Is there a way to get jpm to build and then run that executable straight away? Or do I need to write a wrapper script?

**Comment by ianthehenry** [comment, marked as answer]
Mmm, you could make a custom build rule:

```janet
(declare-executable
  :name "hello"
  :entry "main.janet")

(rule "hello" ["build"]
  (shell "build/hello"))
```

And execute it with `jpm -l run hello`. But I'm not sure if jpm has a nice way to forward command line arguments to your script...

**Comment by m-r-hunt** [comment]
Yeah that works well for me. It's a game/interactive toy made with Jaylib so command line arguments aren't a problem, just running it is fine. I somehow missed the `run` subcommand documentation in `jpm`.

P.S. Ian, your blogposts and book on Janet got me interested in the language, so thanks for that too. 😄

**Reply by ianthehenry** [reply]
Maybe worth noting -- unless you're writing native modules yourself, you don't *have* to compile your project ahead of time. Once you've installed Jaylib you can just `jpm -l janet main.janet` -- the Janet interpreter can dynamically load native dependencies just fine. (Of course, if you've written other native modules in your app, you'll need to recompile them.) (Also the `-l` assumes that you installed jaylib with `-l` as well which I would recommend but which is not the default.)

(Also you're welcome! Welcome to Janet!)
